### ISSUE TITLE
Support Shizuku installation method

### ISSUE THREAD

**Issue by archon810** [android-police/apkmirror-public#174]
See https://github.com/android-police/apkmirror-public/issues/116#issuecomment-774044700.

> I am noting that even though Xiaomi restricts installation of apks using third party installers, `adb install-multiple` still can install app bundles. An app can utilise the similar code but it would need android.permission.INSTALL_PACKAGES permission, which is only granted to system apps. However, there is an alternative which SAI (Split Apk Installer) uses, Shizuku mode which allows SAI to run ADB commands from the mobile device, provided Shizuku is running and SAI has the permission to use it. https://play.google.com/store/apps/details?id=moe.shizuku.privileged.api I wonder if @archon810 is interested to implement it in APK mirror installer. I use Shizuku mode on SAI on my MIUI device with MIUI optimization enabled. And it works, because it's technically a shell based installer.

**Comment by Boy656** [comment]
i am here for requesting the same thing 
adding support for shizuku on apk bundle for xiaomi users

**Comment by VisionR1** [comment]
I'm curious, for Xiaomi users with root and Miui optimization enable, can just install the base and after in the data/app where the base installed copy and paste the other parts (maybe need renamed and  right permission first) and work?

**Comment by Boy656** [comment]
you can use sai spilt application for install the apks application and Zarchive for access the Android/data
sai spilt apks install the application using Shizuka
Zarchive opens Android/obb/data with shizuka on Android 11+

**Comment by VisionR1** [comment]
> you can use sai spilt application for install the apks application and Zarchive for access the Android/data sai spilt apks install the application using Shizuka Zarchive opens Android/obb/data with shizuka on Android 11+

Thanks for answer. Yeah i ask because i'm still in Android 10 Miui 12, and i don't want disable the MIUI Optimization

Edit: 1 method with backup) I try Swift Backup and working
2 without backup) Install the base.apk copy&paste other components (need rename and proper permission and reboot the device)

**Comment by Boy656** [comment]
@VisionR1 me too i don't want to disable the MIUI Optimization so that is why i upgraded my system android to android 11+ because Shizuka make it easier to install spilt apks and faster cope files to android/data and android/obb
if you can install spilt apks in your android 10 miui 12 so you are more than good but really i dont like miui 12 because it is ugly so i prefer miui 12.5 and above because they fixed the error of install spilt apks and install the application in general inside the other application because miui 12 you cant install any application inside the application without disable miui optimization or shizuka enabled

**Comment by VisionR1** [comment]
> @VisionR1 me too i don't want to disable the MIUI Optimization so that is why i upgraded my system android to android 11+ because Shizuka make it easier to install spilt apks and faster cope files to android/data and android/obb if you can install spilt apks in your android 10 miui 12 so you are more than good but really i dont like miui 12 because it is ugly so i prefer miui 12.5 and above because they fixed the error of install spilt apks and install the application in general inside the other application because miui 12 you cant install any application inside the application without disable miui optimization or shizuka enabled

Yeah, I found this 2 method i write and work good, i had update to Android 11 and 12 but the irony is i don't like the MIUI after 12 because the UI for me is ugly with all this new icon, and the limed with Android/data/obb (but this fix with root) and for that go back to Android 10

**Comment by VisionR1** [comment]
> @VisionR1 me too i don't want to disable the MIUI Optimization. So that is why i upgraded my system android to android 11+ because Shizuka make it easier to install spilt apks and faster copy files to android/data and android/obb. if you can install spilt apks in your android 10 miui 12 so you are more than good but really i dont like miui 12 because it is ugly. I prefer miui 12.5 and above because they fixed the error of install spilt apks and install the application in general inside the other application. miui 12 you cant install any application inside the application without disable miui optimization or shizuka enabled.

One question you say have Android 11 MIUI 12.5 with Shizuku, without unlock bootloader and root ?

**Comment by Boy656** [comment]
i were have android 11 with miui 12.5 but now i am using android 12 with miui 14 china version and yeah you can use shizuka from android 11 and above.
if you have android 10 and less you can use shizuka but to enable it you shoud(must) use a pc termnal with your phone connected and use this command (adb shell sh /storage/emulated/0/Android/data/moe.shizuku.privileged.api/start.sh) but at first you should install the application and yes every reboot you should reenable the application again and yes you can enable shizuka using root (magisk and others) as i know from your messages you have root with android 10 and miui 12 so you can give it a try download shizuka from google play store

**Comment by VisionR1** [comment]
> i were have android 11 with miui 12.5 but now i am using android 12 with miui 14 china version and yeah you can use shizuka from android 11 and above. if you have android 10 and less you can use shizuka but to enable it you shoud(must) use a pc termnal with your phone connected and use this command (adb shell sh /storage/emulated/0/Android/data/moe.shizuku.privileged.api/start.sh) but at first you should install the application and yes every reboot you should reenable the application again and yes you can enable shizuka using root (magisk and others) as i know from your messages you have root with android 10 and miui 12 so you can give it a try download shizuka from google play store

Thanks for info, yeah, just yesterday i update to Android 11 MIUI 12.5, not go to 12 MIUI 14 EEA because lot say have many bug
and you have root ?

**Comment by Boy656** [comment]
no not anymore because i founded shizuka

**Comment by VisionR1** [comment]
> no not anymore because i founded shizuka

That's good, and you pass 3/3 in play integrity check?

**Comment by Boy656** [comment]
my bootloader is open
so its obviously pass 1 and fail in the other

you have a root too so your bootloader is unlocked

**Comment by VisionR1** [comment]
> my bootloader is open so its obviously pass 1 and fail in the other
> 
> you have a root too so your bootloader is unlocked

Yeah and just update to Android 12 MIUI in my Xiaomi POCO X3 NFC, now test it

I thought it was possible to have Shizuku with locked Bootloader and pass 3/3

**Comment by VisionR1** [comment]
> @VisionR1 yeah you can have shizuka and pass 3 because it didnt change anything in your system just shizuka like a pc connected to your phone and allow to you to use adb commands as you like without pc

That's is good to know that, Android 12 Stretch overscroll is really bad any way disable only this? with magisk module or normal ?

**Comment by Boy656** [comment]
@VisionR1 if you mean the animation you can disable them
.
just go to 1- additional setting's then 2- accessibility  3-go to vision 4- remove animations

now you will not see any animation when enter an application or when scrolling in application i think maybe will help you

but if you mean the new scrolling in android 12 when you reach to the last of the content in any application in android then this is a smooth thing google added it to android 12 and maybe if you removed the animation you will not see it until you enable the animation again

**Comment by VisionR1** [comment]
> @VisionR1 if you mean the animation you can disable them . just go to 1- additional setting's then 2- accessibility 3-go to vision 4- remove animations
> 
> now you will not see any animation when enter an application or when scrolling in application i think maybe will help you
> 
> but if you mean the new scrolling in android 12 when you reach to the last of the content in any application in android then this is a smooth thing google added it to android 12 and maybe if you removed the animation you will not see it until you enable the animation again

Yeah i try this, but disable all animations, i want only the new scrolling, is bad for my eyes and make me dizzy

**Comment by Boy656** [comment]
@VisionR1 i think you cant but try to disable one by one of the animation in settings developer
you will just disable one only

**Comment by VisionR1** [comment]
> @VisionR1 i think you cant but try to disable one by one of the animation in settings developer you will just disable one only

Yeah i test it, If no solution i go back to A11 MIUI 12.5

**Comment by archon810** [comment]
We have been working on Shizuku and root support for APKMirror Installer, and it should be released shortly if testing goes well.

**Comment by X1nto** [comment]
Shizuku and root installation methods are now available in 1.10.0. Thank you for your patience.
